parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputFormat.java

```diff
@@ -317,6 +319,16 @@ private static int getMaxPaddingSize(Configuration conf) {
     return conf.getInt(MAX_PADDING_BYTES, DEFAULT_MAX_PADDING_SIZE);
   }
 
+  public static ValuesWriterFactory getValuesWriterFactory(Configuration conf) {
+    ValuesWriterFactory factory = ParquetProperties.DEFAULT_VALUES_WRITER_FACTORY;
+
+    if (factory instanceof Configurable) {
+      Configurable configurableFactory = (Configurable) factory;
+      configurableFactory.setConf(conf);
+    }
```

Comment:
Hmm ok, I can remove that piece of code. I'll add a comment in the `ValuesWriterFactory` interface to indicate that if someone wants to use Hadoop config in their factory they need to hook it up in `ParquetOutputFormat` it's not very obvious. 

We've gone back and forth a bit on the desired approach. 
My initial implementation expressed configuration in terms of encodings:
```
parquet.writer.encoding-override.<type> = "encoding1[,encoding2]"
"parquet.writer.encoding-override.int32" = "plain"
```
@isnotinvain suggested the reflection based approach as it was more flexible - can do manual experimentation as well as potentially automated encoding selection(see notes above). 
Either this approach or the reflection based override method works for us right now (I have a subclass of ValuesWriterFactory that reads the encoding for a given type from config in a fork). 

Anyway, I think for now we can go ahead with what we have right now(not exposing any of these in config) after I remove the `Configurable` code. This will help us break up some of the coupling in ParquetProperties. We could discuss which of these approaches are more appealing to various groups of Parquet users and I'd be happy to add a PR. Let me know if this sounds reasonable.